test/go/testing-matchers/main_test.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"testing"
+	"cdk.tf/go/stack/generated/kreuzwerker/docker"
+	"github.com/hashicorp/terraform-cdk-go/cdktf"
+	"github.com/aws/jsii-runtime-go"
+)
+
+var app = cdktf.Testing_App(nil)
+var stack = NewMyStack(app, "stack")
+var synth = cdktf.Testing_Synth(stack)
+var fullSynth = cdktf.Testing_FullSynth(stack)
+
+var stackInvalid = NewMyStackInvalid(app, "stack-invalid")
+var fullSynthInvalid = cdktf.Testing_FullSynth(stackInvalid)
+
+
+func TestToHaveResourcePass(t *testing.T){
+	assertion := cdktf.Testing_ToHaveResource(synth, docker.Container_TfResourceType())
+
+	if !*assertion.Pass() {
+		t.Fail()
+		t.Error(assertion.Message())
```

Comment:
The [docs](https://pkg.go.dev/testing#B.Error) for `t.Fail()` state
> Error is equivalent to Log followed by Fail. 

As I understand it, we can remove the calls to `t.Fail()` and only keep `t.Error(…)` making it shorter 🤓